source/DSCResources/DSC_CMForestDiscovery/DSC_CMForestDiscovery.psm1

```diff
@@ -0,0 +1,299 @@
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    $forestDiscovery = (Get-CMDiscoveryMethod -Name ActiveDirectoryForestDiscovery -SiteCode $SiteCode).Props
+    $enabledStatus   = (($forestDiscovery | Where-Object -FilterScript {$_.PropertyName -eq 'Settings'}).Value1 -eq 'Active')
+    $forestSchedule  = ($forestDiscovery | Where-Object -FilterScript {$_.PropertyName -eq 'Startup Schedule'}).Value1
+    $subnetboundary  = ($forestDiscovery | Where-Object -FilterScript {$_.PropertyName -eq 'Enable Subnet Boundary Creation'}).Value
+    $siteboundary    = ($forestDiscovery | Where-Object -FilterScript {$_.PropertyName -eq 'Enable AD Site Boundary Creation'}).Value
+
+    $convertCimParam = @{
+        ScheduleString = $forestSchedule
+        CimClassName   = 'DSC_CMForestDiscoveryPollingSchedule'
+    }
+    $syncSchedule = ConvertTo-CimCMScheduleString @convertCimParam
+
+    return @{
+        SiteCode                                  = $SiteCode
+        Enabled                                   = $enabledStatus
+        PollingSchedule                           = $syncSchedule
+        EnableActiveDirectorySiteBoundaryCreation = $siteboundary
+        EnableSubnetBoundaryCreation              = $subnetboundary
+    }
+}
+
+ <#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the forest discovery method. If settings is set to $false no other value provided will be
+        evaluated for compliance.
+
+    .PARAMETER PollingSchedule
+        Specifies a schedule and determines how often Configuration Manager attempts to discover Active Directory forest.
+
+    .PARAMETER EnableActiveDirectorySiteBoundaryCreation
+        Indicates whether Configuration Manager creates Active Directory boundaries from AD DS discovery information.
+
+    .PARAMETER EnableSubnetBoundaryCreation
+        Indicates whether Configuration Manager creates IP address range boundaries from AD DS discovery information.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $PollingSchedule,
+
+        [Parameter()]
+        [Boolean]
+        $EnableActiveDirectorySiteBoundaryCreation,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSubnetBoundaryCreation
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+        if ($Enabled -eq $true)
+        {
+            $exludelist = @('Verbose','PollingSchedule')
```

Comment:
I would change this to be an includelist, there are other commonparameters that could be used that will cause an error.

Reply:
Swapped out for an includeList.